shared-libs/transitions/test/unit/transitions/create_user_for_contacts.js

```diff
@@ -588,5 +727,98 @@ describe('create_user_for_contacts', () => {
         });
       });
     });
+
+    it('creates user and replaces user for the same contact at the same time', async () => {
```

Comment:
@dianabarsan your scenario here seems more simple (since there is no `replace` happening).  Even still, I agree that this seems like a plausible scenario that could happen.  Unfortunately, I don't think there is an easy fix to this behavior.  We would need to be able to check and see if there is already an existing user for that contact. Based on similar discussions we had during the replace project, that might mean a new Couch view to query on.  Seems like quite a bit of work just to avoid extra users created in an edge case.  

That being said, my main goal is to have things behave in the _least surprising way._  So, let me know which behavior seems best here!

Reply:
> able to check and see if there is already an existing user for that contact

Off the top of my head, an alternative would be for the users api, that will be used for creating any user, regardless, checks whether the contact has the create new user flag and remove it (the flag)? How messy would the code get?

Reply:
I like this idea!  I have pushed the changes to the branch. Does not seem too messy!